### ISSUE TITLE
libcloud.common.google.GoogleBaseError Invalid Credentials

### ISSUE THREAD

**Issue by sverch** [sverch/butter#32]
I got this error just once:

```
                code = None
            raise ResourceNotFoundError(message, self.status, code)
    
        elif self.status == httplib.BAD_REQUEST:
            if (not json_error) and ('error' in body):
                (code, message) = self._get_error(body)
            else:
                message = body
                code = None
            raise InvalidRequestError(message, self.status, code)
    
        else:
            if (not json_error) and ('error' in body):
                (code, message) = self._get_error(body)
            else:
                message = body
                code = None
>           raise GoogleBaseError(message, self.status, code)
E           libcloud.common.google.GoogleBaseError: {'domain': 'global', 'reason': 'authError', 'message': 'Invalid Credentials', 'locationType': 'header', 'location': 'Authorization'}

.tox/gce/lib/python3.6/site-packages/libcloud/common/google.py:296: GoogleBaseError
```

I should try to reproduce it, for now just filing an issue to see how often it comes up.

**Comment by rauldelperal** [comment]
I also got the same error just once.

An exception occurred during task execution. To see the full traceback, use -vvv. The error was: libcloud.common.google.GoogleBaseError: {u'locationType': u'header', u'domain': u'global', u'message': u'Invalid Credentials', u'reason': u'authError', u'location': u'Authorization'}
2018-11-21 16:05:51,152 p=24546 u=root |  fatal: [127.0.0.1]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_kgHIm0/ansible_module_gce.py\", line 780, in <module>\n    main()\n  File \"/tmp/ansible_kgHIm0/ansible_module_gce.py\", line 739, in main\n    module, gce, inames, number, lc_zone)\n  File \"/tmp/ansible_kgHIm0/ansible_module_gce.py\", line 552, in create_instances\n    instance, lc_machine_type, lc_image(), **gce_args\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/compute/drivers/gce.py\", line 3992, in create_node\n    return self.ex_get_node(name, location.name)\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/compute/drivers/gce.py\", line 7211, in ex_get_node\n    response = self.connection.request(request, method='GET').object\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/compute/drivers/gce.py\", line 121, in request\n    response = super(GCEConnection, self).request(*args, **kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/common/google.py\", line 808, in request\n    *args, **kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/common/base.py\", line 637, in request\n    response = responseCls(**kwargs)\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/common/base.py\", line 155, in __init__\n    self.object = self.parse_body()\n  File \"/usr/local/lib/python2.7/dist-packages/libcloud/common/google.py\", line 296, in parse_body\n    raise GoogleBaseError(message, self.status, code)\nlibcloud.common.google.GoogleBaseError: {u'locationType': u'header', u'domain': u'global', u'message': u'Invalid Credentials', u'reason': u'authError', u'location': u'Authorization'}\n", "module_stdout": "", "msg": "MODULE FAILURE", "rc": 1}

**Comment by sverch** [comment]
Hey @rauldelperal thanks for the stack trace!  Glad to know it's not just me.  Seems like you're getting it in an Ansible library as well.

I've actually had it come up more since I posted this, and the new issue is here: https://github.com/getcloudless/cloudless/issues/18 (this project has moved there actually).

Since this appears to be ultimately a libcloud issue now that I know I'm not the only one, I went ahead and filed https://issues.apache.org/jira/browse/LIBCLOUD-1027.
